src/components/renderer/ReverseGeocode.tsx

```diff
@@ -0,0 +1,156 @@
+        (el) => el.id === element.formattedAddressElementId,
+      )
+    }
+  }, [
+    element.formattedAddressElementId,
+    element.showStreetAddress,
+    formSubmissionModel.elements,
+  ])
+
+  const mergeReverseGeocodeData = React.useCallback(
+    (reverseGeocodeResult: SubmissionTypes.S3SubmissionData['submission']) => {
+      onLookup(({ submission, elements, executedLookups }) => {
+        return {
+          elements,
+          submission: {
+            ...submission,
+            ...reverseGeocodeResult,
+          },
+          executedLookups,
+        }
+      })
+    },
+    [onLookup],
+  )
+
+  React.useEffect(() => {
+    const abortController = new AbortController()
+    const effect = async () => {
+      setReverseGeocodingState({ isReverseGeocoding: true })
+      try {
+        if (coords && formattedAddressElement) {
+          if (element.showStreetAddress && formattedAddressElement) {
+            const reverseGeocodeResult: SubmissionTypes.S3SubmissionData['submission'] =
+              {}
+            switch (formattedAddressElement.type) {
+              case 'text': {
+                const {
+                  reverseGeocodeResult: { formatted_address },
+                } = await formService.getGoogleMapsReverseGeocoding({
+                  lat: coords.latitude,
+                  lng: coords.longitude,
+                  formId: formDefinition.id,
+                  abortSignal: abortController.signal,
+                })
+                reverseGeocodeResult[formattedAddressElement.name] =
+                  formatted_address
+                break
+              }
+              case 'geoscapeAddress': {
+                const { reverseGeocodeResult: geoscapeReverseGeocodeResult } =
+                  await formService.getGeoscapeReverseGeocoding({
+                    lat: coords.latitude,
+                    lng: coords.longitude,
+                    formId: formDefinition.id,
+                    abortSignal: abortController.signal,
+                  })
+                reverseGeocodeResult[formattedAddressElement.name] =
+                  geoscapeReverseGeocodeResult
+                break
+              }
+            }
+            if (!abortController.signal.aborted) {
+              mergeReverseGeocodeData(reverseGeocodeResult)
+            }
+          }
+        }
+      } catch (e) {
+        let errorMsg = 'Could not find your address.'
+        if (e instanceof OneBlinkAppsError && e.httpStatusCode !== 404) {
+          errorMsg = e.message
+        }
+
+        if (isOffline) {
+          errorMsg =
+            'It looks like you&apos;re offline. Please try again when connectivity is restored.'
+        }
+        setReverseGeocodingState((current) => ({
+          ...current,
+          reverseGeocodingErrorMsg: errorMsg,
+        }))
+      }
+      setReverseGeocodingState((current) => ({
+        ...current,
+        isReverseGeocoding: false,
+      }))
```

Comment:
Prob want to have 1 state update in the try and the catch so we aren't doing 2 state updates one after the other in the catch scenario.